framework/src/task_manager/taskmgr_utils.c

```diff
@@ -0,0 +1,81 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <sched.h>
+#include <debug.h>
+#include <sys/types.h>
+#include <apps/builtin.h>
+#include <tinyara/fs/ioctl.h>
+#include <task_manager/task_manager.h>
+#include "task_manager_internal.h"
+
+/****************************************************************************
+ * taskmgr_utils
+ ****************************************************************************/
+void tm_update_task_state(int handle)
+{
+	int ret;
+
+	ret = ioctl(g_taskmgr_fd, TMIOC_CHECK_ALIVE, TASK_PID(handle));
+	if (ret < 0 && TASK_LIST_ADDR(handle) != NULL) {
```

Comment:
ret will be returned OK or ERROR. So I think we'd better check ret != OK instead of ret < 0

Reply:
Because ioctl returns the result value returned by internal ioctl function.
let me consider it more.